server/routes/routes_v1_1.go

```diff
@@ -0,0 +1,62 @@
+package routes
+
+import (
+	"github.com/gin-gonic/gin"
+
+	"github.com/numaproj/numaflow/server/apis/v1_1"
+)
+
+func v1_1Routes(r gin.IRouter) {
+	handler, err := v1_1.NewHandler()
+	if err != nil {
+		panic(err)
+	}
+	// List all namespaces that have Pipeline or InterStepBufferService objects.
+	r.GET("/namespaces", handler.ListNamespaces)
+	// Summarized information of all the namespaces in a cluster wrapped in a list.
+	r.GET("/cluster-summary", handler.GetClusterSummary)
+	// Create a Pipeline.
+	r.POST("/namespaces/:namespace/pipelines", handler.CreatePipeline)
+	// All pipelines for a given namespace.
+	r.GET("/namespaces/:namespace/pipelines", handler.ListPipelines)
+	// Get a Pipeline information.
+	r.GET("/namespaces/:namespace/pipelines/:pipeline", handler.GetPipeline)
+	// Get a Pipeline health information.
+	r.GET("/namespaces/:namespace/pipelines/:pipeline/health", handler.GetPipelineStatus)
+	// Update a Pipeline.
+	r.PUT("/namespaces/:namespace/pipelines/:pipeline", handler.UpdatePipeline)
+	// Delete a Pipeline.
+	r.DELETE("/namespaces/:namespace/pipelines/:pipeline", handler.DeletePipeline)
+	// Patch the pipeline spec to achieve operations such as "pause" and "resume".
+	r.PATCH("/namespaces/:namespace/pipelines/:pipeline", handler.PatchPipeline)
+	// Create an InterStepBufferService object.
+	r.POST("/namespaces/:namespace/isb-services", handler.CreateInterStepBufferService)
+	// List all the InterStepBufferService objects for a given namespace.
+	r.GET("/namespaces/:namespace/isb-services", handler.ListInterStepBufferServices)
+	// Get an InterStepBufferService object.
+	r.GET("/namespaces/:namespace/isb-services/:isb-services", handler.GetInterStepBufferService)
+	// Update an InterStepBufferService object.
+	r.PUT("/namespaces/:namespace/isb-services/:isb-services", handler.UpdateInterStepBufferService)
+	// Delete an InterStepBufferService object.
+	r.DELETE("/namespaces/:namespace/isb-services/:isb-services", handler.DeleteInterStepBufferService)
+	// Get all the Inter-Step Buffers of a pipeline.
+	r.GET("/namespaces/:namespace/pipelines/:pipeline/isbs", handler.ListPipelineBuffers)
+	// Get all the watermarks information of a pipeline.
+	r.GET("/namespaces/:namespace/pipelines/:pipeline/watermarks", handler.GetPipelineWatermarks)
+	// Update a vertex spec.
+	r.PUT("/namespaces/:namespace/pipelines/:pipeline/vertices/:vertex", handler.UpdateVertex)
+	// Get all the vertex metrics of a pipeline.
+	r.GET("/namespaces/:namespace/pipelines/:pipeline/vertices/metrics", handler.GetVerticesMetrics)
+	// Get all the pods of a vertex.
+	r.GET("/namespaces/:namespace/pipelines/:pipeline/vertices/:vertex/pods", handler.ListVertexPods)
+	// Get the metrics such as cpu, memory usage for a pod.
+	r.GET("/metrics/namespaces/:namespace/pods/", handler.ListPodsMetrics)
```

Comment:
Is there a reason that the "logs" and "events" resources are put under the namespace / pod hierarchy, but the "metrics" is a top level resource? Should they follow the same route patten? such as
```
r.GET("/namespaces/:namespace/pods/metrics", handler.ListPodsMetrics)
```

Reply:
This is following similar patten of k8s pod metrics API, where `metrics.k8s.io` is a different API group. 

`/apis/metrics.k8s.io/v1beta1/namespaces/:namespace/pods`.

We don't need to have a API group but add `metrics` to indicate it's in a different sub route (other than `namespace/pipeline/vertex`).

The reason why k8s has this API patten is, the metrics api group also supports other metrics , for example node metrics.

`/apis/metrics.k8s.io/v1beta1/nodes`

Reply:
BTW - @jy4096 @kohlisid, please remove the tailing slash.